Lib/test/test_ttk/test_widgets.py

```diff
@@ -479,11 +490,12 @@ def _show_drop_down_listbox(self):
         x, y = width - 5, 5
         if sys.platform != 'darwin':  # there's no down arrow on macOS
             self.assertRegex(self.combo.identify(x, y), r'.*downarrow\Z')
-        self.combo.event_generate('<ButtonPress-1>', x=x, y=y)
+        self.combo.event_generate('<Button-1>', x=x, y=y)
         self.combo.event_generate('<ButtonRelease-1>', x=x, y=y)
-        self.combo.update_idletasks()
 
     def test_virtual_event(self):
+        if tk_version >= (9, 0) and sys.platform == 'darwin':
+            self.skipTest('test hangs due to Tk 9.0 bug 8b49e9cfa6')
```

Comment:
I can add a link.  But it turns out not to be a bug, exactly.  There is a change in Tk 9, which is that the combobox menu became a native macOS menu.  It looks much better and it works fine in actual human interaction with the combobox.  But the act of launching a macOS menu causes Apple to create a special event loop which does not pass events to the Tk client until the user dismisses the menu with a mouse click, which may or may not select a menu item.  While this is not an issue with using a combobox, it does create an issue with trying to use generated Tk events to open and close the combobox, as that test does.  The event generate events are received and handled by Tk, but Tk currently has no way to dismiss a native macOS menu.  So it has to wait until Apple decides to dismiss the menu based on user interaction.

The choice of not having a "menu unpost" command in the macOS and Window ports of Tk was made intentionally, because those OS's provide their own (user interaction based) ways of dismissing a menu.  X11 provides an unpost command, so Tk wraps that.  In fact, macOS could support a "menu unpost" command.  But that would require a TIP since it would be a change in the Tk API.  If there were such a command then it could be bound to a Tk event and that would allow posting and dismissing combobox menus with event generate commands.

As usual, things that seem simple on the surface can be complex.

Anyway, I think I am going to close the Tk ticket as a "won't fix right now" and consider whether to write a TIP. to add a menu unpost command for Aqua.  I think it just means that that one test will have to be skipped for Tk 9.  I don't see any immediate workaround.